packages/nimble-components/src/wafer-map/index.ts

```diff
@@ -0,0 +1,67 @@
+import { attr, observable } from '@microsoft/fast-element';
+import { DesignSystem, FoundationElement } from '@microsoft/fast-foundation';
+import { template } from '../theme-provider/template';
+import { styles } from './styles';
+import {
+    WaferMapColorsScale,
+    WaferMapDie,
+    WaferMapOrientation,
+    WaferMapQuadrant
+} from './types';
+
+declare global {
+    interface HTMLElementTagNameMap {
+        'nimble-wafer-map': WaferMap;
+    }
+}
+
+/**
+ * A nimble-styled WaferMap
+ */
+export class WaferMap extends FoundationElement {
+    @attr({
+        attribute: 'quadrant'
+    })
+    public quadrant: WaferMapQuadrant = WaferMapQuadrant.topLeft;
+
+    @attr({
+        attribute: 'orientation'
+    })
+    public orientation: WaferMapOrientation = WaferMapOrientation.top;
+
+    @attr({
+        attribute: 'max-characters'
+    })
+    public maxCharacters = 4;
+
+    @attr({
+        attribute: 'show-die-labels'
+    })
+    public showDieLabels = true;
+
+    @attr({
+        attribute: 'die-labels-suffix'
+    })
+    public dieLabelsSuffix = '';
+
+    @attr({
+        attribute: 'is-continuous'
+    })
+    public isContinuous = true;
+
+    @attr({
+        attribute: 'highlighted-values'
+    })
+    public highlightedValues: number[] = [];
+
+    @observable public dies: WaferMapDie[] = [];
+    @observable public colorsScale: WaferMapColorsScale = {} as WaferMapColorsScale;
```

Comment:
We should not fight the type system using the `as` keyword here. The current type described says that you must provide a colors and values array.

I'd suggest that the behavior be that the default colors and values arrays are empty. And if both arrays are empty then internally the wafer-map should fallback to the default colors scale.

```suggestion
    @observable public colorsScale: WaferMapColorsScale = {
        colors: [],
        values: []
    }
```

Another option I would NOT recommend is making the type itself looser by allowing colors and values to be undefined:

```ts
export interface WaferMapColorsScale {
    colors?: string[];
    values?: number[];
}
```

And you could go even further and make the property itself optional:

```ts
@observable public colorsScale?: WaferMapColorsScale;
```

But I would strongly discourage going that approach. With strict type checking you have the opportunity to avoid adding lots of unnecessary null checks to your code by preventing these values from being optional.